test/syncedEnforcer.test.ts

```diff
@@ -0,0 +1,20 @@
+import { Enforcer, newEnforcer, newModel, newSyncedEnforcer } from '../src';
+
+async function testEnforce(e: Enforcer, sub: any, obj: string, act: string, res: boolean): Promise<void> {
+  expect(e.enforceSync(sub, obj, act)).toBe(res);
+}
+
+test('TestGFunctionInSyncedEnforcer', async () => {
+  const m = newModel();
+  m.addDef('r', 'r', 'sub, obj, act');
+  m.addDef('p', 'p', 'sub, obj, act');
+  m.addDef('g', 'g', '_, _');
+  m.addDef('e', 'e', 'some(where (p.eft == allow))');
+  m.addDef('m', 'm', 'g(r.sub, p.sub) && r.obj == p.obj && r.act == p.act');
+
+  const e = await newSyncedEnforcer(m);
+
+  await e.addPermissionForUser('alice', 'data1', 'invalid');
```

Comment:
Hmm. This test case could be improved a little bit.

Usually synced enforcer fails when there is a access right to an object, but not for the user in question.

Maybe `e.addPermissionForUser('bob', 'data1', 'read')` instead.

This can happen with domains as well. i.e. there is an access right, but not for the domain in question.

Let's say that `alice` has access right to all files in `domain2`, but only read access to data1 in `domain1`. Before synced adapter would fail to recognize this and just assume that `domain2` rights apply to `domain1` as well.